test/jdk/java/io/File/createTempFile/SpecialTempFile.java

```diff
@@ -84,6 +87,30 @@ private static void test(String name, String[] prefix, String[] suffix,
         }
     }
 
+    private static int windowsBuild() throws IOException {
+        if (!OperatingSystem.isWindows())
+            throw new UnsupportedOperationException();
+
+        String cmd = "Systeminfo";
+        Process p = Runtime.getRuntime().exec(new String[] {cmd});
```

Comment:
I'm not a big fan of running third-party tools in our tests; did you try fixing the regex expression?
Replace: `"^.*[11|2025]$"` (character class) -> `"^.*(11|2025)$"` (alternative strings)

Reply:
No, I replaced it with splitting the OS version string. In any case, would that cover, e.g., the hypothetical cases Windows 27, Windows Server 2038, etc.?

Unfortunately Miscrosoft does not provide a reliable API to obtain the build number. `GetVersionEx` is deprecated. [KUSER_SHARED_DATA](https://learn.microsoft.com/en-us/windows-hardware/drivers/ddi/ntddk/ns-ntddk-kuser_shared_data) contains `NtBuildNumber` but some device driver kit has to be installed to use it.

Reply:
No, it wouldn't cover any future Windows versions, but it would definitely buy us a few years.

Regarding a more permanent solution: the test is supposed to verify the fix for [JDK-8013827](https://bugs.openjdk.org/browse/JDK-8013827). I think it should accept both outcomes (success and the specific exception); the only unacceptable outcomes are:
- timeout,
- unexpected exception.

We don't need to know the exact Windows version to verify that the test doesn't time out.

Reply:
> I think it should accept both outcomes (success and the specific exception)

But whether there's an exception varies by Windows version / build number. I'll take another look and see whether there is some other way.

Reply:
we don't care if there's an exception or not. It's not part of the spec.

Reply:
Yes, I think I'll rework this. Thanks.

Reply:
The build number could be obtained from the value of the registry key
```
HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\CurrentBuildNumber
```
but it seems not to be worth the trouble. In 51336af I instead changed the test of reserved names on Windows to ignore any `IOException` thrown by `File.createTempFile`. The change was verified on the aforementioned Windows variants and on Linux and macOS.